lib/flip_transition.dart

```diff
@@ -0,0 +1,182 @@
+  late CardSide _currentSide;
+
+  @override
+  void initState() {
+    super.initState();
+    _currentSide = _getSideFor(widget.animation.status);
+    widget.animation.addStatusListener(_handleChange);
+  }
+
+  @override
+  void didUpdateWidget(covariant FlipTransition oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.animation != oldWidget.animation) {
+      oldWidget.animation.removeStatusListener(_handleChange);
+      widget.animation.addStatusListener(_handleChange);
+      _currentSide = _getSideFor(widget.animation.status);
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    widget.animation.removeStatusListener(_handleChange);
+  }
+
+  CardSide _getSideFor(AnimationStatus status) {
+    switch (status) {
+      case AnimationStatus.dismissed:
+      case AnimationStatus.reverse:
+        return CardSide.front;
+      case AnimationStatus.forward:
+      case AnimationStatus.completed:
+        return CardSide.back;
+    }
+  }
+
+  void _handleChange(AnimationStatus status) {
+    final newSide = _getSideFor(status);
+    if (newSide != _currentSide) {
+      setState(() {
+        _currentSide = newSide;
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final frontPositioning = widget.fill == Fill.front ? _fill : _noop;
+    final backPositioning = widget.fill == Fill.back ? _fill : _noop;
+
+    return Stack(
+      alignment: widget.alignment,
+      fit: StackFit.passthrough,
+      children: <Widget>[
+        frontPositioning(_buildContent(child: widget.front)),
+        backPositioning(_buildContent(child: widget.back)),
+      ],
+    );
+  }
+
+  Widget _buildContent({required Widget child}) {
+    final isFront = child == widget.front;
+    final showingFront = _currentSide == CardSide.front;
+
+    /// pointer events that would reach the backside of the card should be
+    /// ignored
+    return IgnorePointer(
+      /// absorb the front card when the background is active (!isFront),
+      /// absorb the background when the front is active
+      ignoring: isFront ? !showingFront : showingFront,
+      child: AnimationCard(
+        animation: isFront
+            ? (widget.frontAnimator ?? FlipTransition.defaultFrontAnimator)
+                .animate(widget.animation)
+            : (widget.backAnimator ?? FlipTransition.defaultBackAnimator)
+                .animate(widget.animation),
+        direction: widget.direction,
+        child: child,
+      ),
+    );
+  }
+}
+
+class AnimationCard extends AnimatedWidget {
+  const AnimationCard({
+    super.key,
+    required this.child,
+    required this.animation,
+    required this.direction,
+  }) : super(listenable: animation);
+
+  final Animation<double> animation;
+  final Widget child;
+  final Axis direction;
+
+  @override
+  Widget build(BuildContext context) {
+    final transform = Matrix4.identity();
+    transform.setEntry(3, 2, 0.001);
```

Comment:
Where do these magic numbers come from? Why exactly `0.001`? Is any small number ok? Why not exactly 0? Why not `double.minPositive`? Please add comment.

Reply:
The number was already present in the code, I tried it with a couple of numbers including `double.minPositive` but it doesn't give the desired effect.